src/auth/__tests__/index.spec.js

```diff
@@ -693,4 +693,247 @@ describe('auth', () => {
+        'https://samples.auth0.com/oauth/token',
+        unexpectedError,
+      );
+      expect.assertions(1);
+      await expect(auth.loginWithOOB(parameters)).rejects.toMatchSnapshot();
+    });
+
+    it('should handle malformed OOB code', async () => {
+      fetchMock.postOnce(
+        'https://samples.auth0.com/oauth/token',
+        malformedOOBError,
+      );
+      expect.assertions(1);
+      await expect(auth.loginWithOOB(parameters)).rejects.toMatchSnapshot();
+    });
+
+    it('should handle success without binding code', async () => {
+      fetchMock.postOnce('https://samples.auth0.com/oauth/token', success);
+      expect.assertions(1);
+      await expect(auth.loginWithOOB(parameters)).resolves.toMatchSnapshot();
+    });
+
+    it('should handle success with binding code', async () => {
+      parameters.bindingCode = '1234';
+      fetchMock.postOnce('https://samples.auth0.com/oauth/token', success);
+      expect.assertions(1);
+      await expect(auth.loginWithOOB(parameters)).resolves.toMatchSnapshot();
+    });
+  });
+
+  describe('Recovery Code flow', () => {
+    const parameters = {
+      mfaToken: '123',
+      recoveryCode: '123',
+    };
+
+    const success = {
+      accessToken: '1234',
+      expiresIn: 86400,
+      idToken: 'id-123',
+      scope: 'openid profile email address phone',
+      tokenType: 'Bearer',
+    };
+
+    const unAuthorizedClientError = {
+      status: 403,
+      body: {
+        name: 'unsupported_challenge_type',
+        error: 'unsupported_challenge_type',
+        error_description: 'User does not have a recovery-code.',
+      },
+      headers: {'Content-Type': 'application/json'},
+    };
+
+    it('should require MFA Token and Recovery Code', async () => {
+      expect.assertions(1);
+      expect(() =>
+        auth.loginWithRecoveryCode({}),
+      ).toThrowErrorMatchingSnapshot();
+    });
+
+    it('when user does not have Recovery Code', async () => {
+      fetchMock.postOnce(
+        'https://samples.auth0.com/oauth/token',
+        unAuthorizedClientError,
+      );
+      expect.assertions(1);
+      await expect(
+        auth.loginWithRecoveryCode(parameters),
+      ).rejects.toMatchSnapshot();
+    });
+
+    it('should handle unexpected error', async () => {
+      fetchMock.postOnce(
+        'https://samples.auth0.com/oauth/token',
+        unexpectedError,
+      );
+      expect.assertions(1);
+      await expect(
+        auth.loginWithRecoveryCode(parameters),
+      ).rejects.toMatchSnapshot();
+    });
+
+    it('when Recovery code succeeds', async () => {
+      fetchMock.postOnce('https://samples.auth0.com/oauth/token', success);
+      expect.assertions(1);
+      await expect(
+        auth.loginWithRecoveryCode(parameters),
+      ).resolves.toMatchSnapshot();
+    });
+  });
+
+  describe('Multifactor Challenge Flow', () => {
+    const parameters = {
+      mfaToken: '123',
+      challengeType: '123',
+      authenticatorId: '123',
```

Comment:
Please add test cases that verify that `challengeType` and `authenticatorId` are optional.

Reply:
Similar to what was done for `bindingCode` above.

Reply:
Added these changes 👍